framework/contrib/hit/parse.h

```diff
@@ -444,6 +446,95 @@ void merge(Node * from, Node * into);
 /// returned node is the root of the exploded tree.
 Node * explode(Node * n);
 
+// Formatter is used to automatically format hit syntax/input to be uniform in a specified style.
+// After creating a formatter object and configuring it as desired by modifying/calling its members
+// you use the "format" function to format hit text as desired.
+class Formatter : public Walker
+{
+public:
+  /// Constructs a formatter that formats hit text in a canonical default style.
+  Formatter();
+
+  /// Constructs a formatter that formats hit text according to the configuration parameters
+  /// specified in hit_config - which is the text of a hit document of the following form:
+  ///
+  ///     [format]
+  ///       # these parameters set the correspondingly named Formatter member variables.  See them
+  ///       # for detailed descriptions of what they do.
+  ///       indent_string = "  "
+  ///       line_length = 100
+  ///       canonical_section_markers = true
+  ///
+  ///       # This section specifies parameter/section sorgin order as provided by the formatter's
+  ///       # addPattern member function. See the docs for that function for more details.
+  ///       [sorting]
+  ///         # specifies that all level-two sections should have the "first" field first.
+  ///         [pattern]
+  ///           # regex to match section(s) in which to order parameters.  The regex must match the
+  ///           # complete section name (including parent path sections - i.e. the regex to match bar
+  ///           # "[foo] [bar] [][]" must match "foo/bar".
+  ///           section = "[^/]+/[^/]+"
+  ///           # paramter order for matched sections (space-separated regexes). The regexes only
+  ///           # have to partial-match the field names.
+  ///           order = "first"
+  ///         []
+  ///         # specifies that the "foo" section
+  ///         [pattern]
+  ///           section = "foo"
+  ///           # use "**" (at most once) to glob/represent unordered fields
+  ///           order = "bar ** hello"
+  ///         []
+  ///       []
+  ///     []
+  ///
+  /// All fields are optional and the sorting section is also optional.  If the sorting section
+  /// is present, you can have as many patterns as you like, but each pattern section must have
+  /// one set of "section" and "order" fields.
+  Formatter(const std::string & fname, const std::string & hit_config);
+
+  /// Formats the given input hit text (using fname to report better syntax errors) and returns
+  /// the text formatted as specified by the formatter's configuration.
+  std::string format(const std::string & fname, const std::string & input);
+  /// Add a sorting pattern to the formatter.  section is a regex that must match a section's
```

Comment:
Pedantic and minor: Maybe add some whitespace between your comment blocks and the subsequent method names?